src/application.ts

```diff
@@ -158,6 +158,10 @@ export class Application {
    * @private
    */
   public async auth (id?: number, log = this.log): Promise<GitHubAPI> {
+    if (process.env.GHE_HOST && /^https?:\/\//.test(process.env.GHE_HOST)) {
+      throw new Error('Your \`GHE_HOST\` environment variable should not begin with https:// or http://')
+    }
```

Comment:
Sorry, late to the party here, but would be good to only check this on app startup since it's not likely to change at runtime.

Long term, I think the solution would be to extract a `Configuration` class that encapsulates all the config logic

Reply:
> Sorry, late to the party here, but would be good to only check this on app startup since it's not likely to change at runtime.

If you want to open a PR then 👍 otherwise I should get to it soon-ish